lib/vrl/compiler/src/expression/op.rs

```diff
@@ -162,6 +162,14 @@ impl Expression for Op {
             // ... != ...
             Eq | Ne => lhs_def.merge(rhs_def).boolean(),
 
+            // "b" >  "a"
+            // "a" >= "a"
+            // "a" <  "b"
+            // "b" <= "b"
+            Gt | Ge | Lt | Le if lhs_def.is_bytes() && rhs_def.is_bytes() => {
+                lhs_def.scalar(K::Boolean)
```

Comment:
rhs can still be fallible, so we need to merge it with lhs:

```suggestion
                lhs_def.merge(rhs_def).boolean()
```

Reply:
It would be good to add a test that captures this condition.